File: lista_de_exercicios/questao9/questao9.go
package main

import "fmt"

type Imprimivel interface {
    Imprimir()
}

type Relatorio struct{}

func (r Relatorio) Imprimir() {
    fmt.Println("Imprimindo Relatório")
}

type Contrato struct{}

func (c Contrato) Imprimir() {
    fmt.Println("Imprimindo Contrato")
}

File: lista_de_exercicios/questao9/questao9.java
package lista_de_exercicios.questao9;

interface Imprimivel {
    void imprimir();
}

class Relatorio implements Imprimivel {
    public void imprimir() {
        System.out.println("Imprimindo Relatório");
    }
}

class Contrato implements Imprimivel {
    public void imprimir() {
        System.out.println("Imprimindo Contrato");
    }
}
